groups/reconcile.go

```diff
@@ -321,20 +408,44 @@ func updateGroupSettingsToAllowExternalMembers(srv *groupssettings.Service, grou
 		return fmt.Errorf("unable to retrieve group info for group %s: %v", groupEmailId, err)
 	}
 
+	// We will remove any additional members, so there's no point in allowing invitations / adds
 	settings := &groupssettings.Groups{
-		AllowExternalMembers: "true",
-		WhoCanJoin:           "INVITED_CAN_JOIN",
-		WhoCanViewMembership: "ALL_MEMBERS_CAN_VIEW",
-		WhoCanViewGroup:      "ALL_MEMBERS_CAN_VIEW",
-		WhoCanDiscoverGroup:  "ALL_MEMBERS_CAN_DISCOVER",
+		AllowExternalMembers:  "true",
+		WhoCanJoin:            "INVITED_CAN_JOIN",
+		WhoCanViewMembership:  "ALL_MEMBERS_CAN_VIEW",
+		WhoCanViewGroup:       "ALL_MEMBERS_CAN_VIEW",
+		WhoCanDiscoverGroup:   "ALL_MEMBERS_CAN_DISCOVER",
+		WhoCanInvite:          "NONE_CAN_INVITE",
+		WhoCanAdd:             "NONE_CAN_ADD",
+		WhoCanApproveMembers:  "NONE_CAN_APPROVE",
+		WhoCanModifyMembers:   "NONE",
+		WhoCanModerateMembers: "NONE",
 	}
 
-	// We will remove any additional members, so there's no point in allowing invitations / adds
-	settings.WhoCanInvite = "NONE_CAN_INVITE"
-	settings.WhoCanAdd = "NONE_CAN_ADD"
-	settings.WhoCanApproveMembers = "NONE_CAN_APPROVE"
-	settings.WhoCanModifyMembers = "NONE"
-	settings.WhoCanModerateMembers = "NONE"
+	for key, value := range groupSettings {
```

Comment:
wouldn't it be easier to unmarshall into a struct?

Reply:
can fix this in a subsequent PR